gpii/node_modules/settingsHandlers/src/settingsHandlerUtilities.js

```diff
@@ -338,6 +338,16 @@ gpii.settingsHandlers.writeFile = function (content, options) {
 gpii.settingsHandlers.handleFileSolutionEntry = function (solutionEntry, modifier, parser, isWrite) {
     var options = solutionEntry.options;
 
+    var initialContent = options.initialContent;
+    var createFile = options.createFile;
+    //create file, if file does not exist, and set initial content of the file
+    if (createFile) {
+        var exists = fs.existsSync(options.filename);
+        if (!exists) {
+            fs.writeFileSync(options.filename, initialContent, options.encoding || "utf-8");
```

Comment:
In fact, a much more efficient and consistent strategy would be to not attempt to write to the filesystem at all if a read is required from a nonexistent file. We should simply check whether the file exists, and if not, quietly return undefined (there's no need for the "initialContent" option since this is a behaviour common to all settings handlers).